dev-portal/src/services/api-catalog.js

```diff
@@ -1,144 +1,183 @@
-import { getApiGatewayClient } from './api'
-export let subscriptions
-let catalog
-
-export function getCatalog() {
-  if (catalog) return Promise.resolve(catalog)
-
-  return fetchCatalog()
-  .then(({data}) => {
-    catalog = data
-    return data
-  })
-}
-
-export function getApi(apiId) {
-  return getCatalog()
-  .then(() => {
-    let _api
-
-    catalog.forEach(c => {
-      if (_api) return
-      _api = c.apis.find(a => a.id === apiId)
-    })
-
-    return _api
-  })
+import { apiGatewayClient } from './api'
+import { store } from './state'
+
+/* Catalog and API Utils */
+
+/**
+ * 
+ * Does all operations to get user data at once.
+ * 
+ * @param {Boolean} bustCache=true   Ignore the cache and re-make the calls? Defaults to true.
+ */
+export function updateAllUserData(bustCache = true) {
+  return Promise.all([
+    updateCatalogAndApisList(bustCache),
+    updateSubscriptions(bustCache),
+    updateApiKey(bustCache)
+  ])
 }
 
-export function fetchCatalog() {
-  return getApiGatewayClient().then(apiGatewayClient => {
-    return apiGatewayClient.get('/catalog', {}, {}, {})
-  })
+/**
+ * 
+ * Update the catalog for the current user. Both request and response are cached, so unless the cache is busted, this should only ever make one network call.
+ * 
+ * @param {Boolean} [bustCache=false]   Ignore the cache and re-make the network call. Defaults to false.
+ * 
+ */
+export function updateCatalogAndApisList(bustCache = false) {
+  let catalogOrPromise = store.catalog.length ? store.catalog : catalogPromiseCache
+  if (!bustCache && catalogOrPromise) return Promise.resolve(catalogOrPromise)
+
+  return catalogPromiseCache = apiGatewayClient()
+    .then(apiGatewayClient => apiGatewayClient.get('/catalog', {}, {}, {}))
+    .then(({ data = [] }) =>  store.catalog = data)
```

Comment:
Do we need to return here? If so, can we avoid returning assignments? Can produce unexpected behaviors.

https://eslint.org/docs/rules/no-return-assign

Reply:
Yeah, we do. The return assign here (and in a few other places) is intentional. I prefer this b/c it's cleaner, but if I can split it up if you prefer.

Reply:
Are you okay with the `except parens` caveat of that rule? i.e. wrap all return-assigns with parens that you intend?